### PR TITLE
Improve tutorial and README

### PR DESCRIPTION
**Pull request by mattdahl**
This PR makes several improvements to our README.

First, re #94, it adds to the top a clear list of the four core functions of eyecite: extraction, aggregation (resolution), annotation, and cleaning.

Second, re #95, it clarifies the tutorial regarding the hypothetical "database" and other confusing items. All the code in the tutorial is now definitely run-able by the reader, except for two sections that offer psuedo-code for expanded functionality (the `resolutions2` and `annotations2` sections). I also attempted to clarify those sections themselves by providing more details about how one might integrate with an external database or API. It should hopefully be clear that eyecite works fine without doing that, but the pseudo-code should still be helpful for users trying to exploit that functionality.

This PR shouldn't be merged before #96 since I added some links (that don't work) to the new API documentation. This also doesn't address all the open issues re the README; some things (re tests and bigger examples) may still need to be added.

### CODE REVIEW

**Comment by mlissner** [comment]
Matt, you want to take a look at rebasing this and resolving conflicts now that #96 is in?

**Comment by mattdahl** [comment]
Just rebased and added real links to the new API documentation.

File: README.rst [hunk 1]
```diff
@@ -84,137 +88,171 @@ Here's a short example of extracting citations and their metadata from text::
...
+    #    <p><a href='/some_api?cite=5 U. S. 137'>5 <i>U. S.</i> 137</a>, <a href='/some_api?cite=1 Cranch 137'>1 Cranch 137</a>.<p>
+    #    <p><a href='/some_api?cite=1 Cranch 137'>Id.</a></p>
+    #    <p><a href='/some_api?cite=Mass. Gen. Laws ch. 1, § 2'>Mass. Gen.    Laws ch. 1, § 2</a>.</p>
     # """
 
 Ta da!
 
-Getting Citations
+Documentation
 =================
```
Comment (mlissner) [inline README.rst:239]: This needs to be the same length as the title above it to work properly.

**Review by mlissner — CHANGES_REQUESTED** [review]
This is good. One small comment, but you took out the reference to Harvard. Can you get that back in?

File: README.rst (hunk 1 above)
Comment (mattdahl) [reply README.rst]: Actually, I think it's fine as long as it's *at least as long as* the title, but I've changed it to be consistent.

**Comment by mattdahl** [comment]
> This is good. One small comment, but you took out the reference to Harvard. Can you get that back in?

Yes -- restored at the very top!

File: README.rst (hunk 1 above)
Comment (mlissner) [reply README.rst]: Weird! Well, thanks for tweaking anyway.

**Pull request merged**